jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/ssl/SslContextFactory.java

```diff
@@ -1614,23 +1614,23 @@ public void setHostnameVerifier(HostnameVerifier hostnameVerifier)
      * Returns the password object for the given realm.
      *
      * @param realm the realm
-     * @return the Password object
+     * @return the Credential object
      */
-    protected Password getPassword(String realm)
+    protected Credential getPassword(String realm)
```

Comment:
We can't really change the API like that in 12.0.x (actually probably not even in 12.1.x unless we deprecated some time ago).   However, I think there is an easy way around that:
```java
    protected Password getPassword(String realm)
    {        
        String password = System.getProperty(realm);
        return password == null ? null : newPassword(password);
    }

    protected Credential getCredential(String realm)
    {
        return getPassword(realm);
    }

    protected Password netPassword(String password)
    {        
        return new Password(password);
    }

    protected Credential newCredential(String credential)
    {
        return netPassword(credential);
    }
```

Reply:
Your suggested code undoes my change, but I think I get what you're saying. I will create `getCredential` and `newCredential` and revert my changes to the protected/public `getPassword` and `newPassword` methods. Should I mark them deprecated, or leave that to you?

For example:
```java
    /**
     * Creates a new Password object.
     *
     * @param password the password string
     * @return the new Password object
     * @deprecated use {#link newCredential} instead.
     */
    @Deprecated(since = "12.0.13", forRemoval = true)
    public Password newPassword(String password)
    {
        return new Password(password);
    }
```